src/lib/storage.js

```diff
@@ -80,10 +81,18 @@ export const account = {
   },
 
   generate (secret) {
+    const keys = wallet.generate();
+    const sk = keys.getPrivateKeyString();
+    const deviceAddress = keys.getChecksumAddressString();
```

Comment:
How fast is wallet generation?

The only concern I see here is that these 3 consts are processed outside `Promise` and could block UI. Wallet generator is reasonably fast, compared to wallet encryption so it might not be a big problem. 

Can you confirm UI is smooth in account generation form?